lib/V8/v8-utils.cpp

```diff
@@ -202,6 +202,86 @@ static void CreateErrorObject(v8::Isolate* isolate, int errorNumber,
   }
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// @brief set a point in time after which we will abort external connection
+////////////////////////////////////////////////////////////////////////////////
+static std::atomic<double> connectionToBeDeadAt(0.0);
+void setConnectionToBeDeadInMS(uint64_t timeout) {
+  if (timeout == 0) {
+    connectionToBeDeadAt = 0.0;
+  } else {
+    connectionToBeDeadAt = TRI_microtime() + timeout / 1000;
+  }
+}
+
+static void JS_SetConnectionsToBeDeadIn(v8::FunctionCallbackInfo<v8::Value> const& args) {
+  TRI_V8_TRY_CATCH_BEGIN(isolate);
+  v8::HandleScope scope(isolate);
+
+  // extract arguments
+  if (args.Length() != 1) {
+    TRI_V8_THROW_EXCEPTION_USAGE("SetConnectionsToBeDeadIn(<timeout>)");
+  }
+
+  auto when = connectionToBeDeadAt.load();
+  auto now = TRI_microtime();
+
+  double n = TRI_ObjectToDouble(isolate, args[0]);
+  setConnectionToBeDeadInMS(n);
+
+  TRI_V8_RETURN_BOOL((when != 0.0) && (now - when > 0.0) );
+  TRI_V8_TRY_CATCH_END
+}
+
+bool isConnectionToBeDead(v8::FunctionCallbackInfo<v8::Value> const& args) {
+  auto when = connectionToBeDeadAt.load();
+  if (when == 0.0) {
+    return false;
+  }
+  auto now = TRI_microtime();
+  if (now < when) {
+    return false;
+  }
+
+  auto isolate = args.GetIsolate();
+  TRI_CreateErrorObject(isolate, TRI_ERROR_DISABLED, "Timeout for remote!", true);
+  return true;
+}
+
+double getMaxTimeoutConnectionToBeDead(double timeout) {
```

Comment:
Do we really need 2 versions of the functions?

Reply:
as long as js_download works with `double` seconds as a timeout, we need this other function. 
Removed Polimorphic naming, as they do different things.